aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,794 @@
+    }
+
+    /// Generates a unique bridge transfer ID based on transfer details and nonce.
+    ///
+    /// @param details The bridge transfer details.
+    /// @return The generated bridge transfer ID.
+    public(friend) fun bridge_transfer_id(initiator: address, recipient: EthereumAddress, amount: u64, nonce: u64) : vector<u8> {
+        let combined_bytes = vector::empty<u8>();
+        vector::append(&mut combined_bytes, bcs::to_bytes(&initiator));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&recipient));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&amount));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&nonce));
+        keccak256(combined_bytes)
+    }
+    
+    #[view]
+    /// Gets `bridge_transfer_id` from `nonce`.
+    /// @param nonce The nonce of the bridge transfer.
+    /// @return The bridge transfer ID.
+    /// @abort If the nonce is not found in the smart table.
+    public fun get_bridge_transfer_id_from_nonce(nonce: u64): vector<u8> acquires SmartTableWrapper {
+        let table = borrow_global<SmartTableWrapper<u64, vector<u8>>>(@aptos_framework);
+        
+        // Check if the nonce exists in the table
+        assert!(smart_table::contains(&table.inner, nonce), ENONCE_NOT_FOUND);
+
+        // If it exists, return the associated bridge_transfer_id
+        *smart_table::borrow(&table.inner, nonce)
+    }
+
+    #[view]
+    /// Gets incoming `nonce` from `bridge_transfer_id`
+    /// @param bridge_transfer_id The ID bridge transfer.
+    /// @return the nonce
+    /// @abort If the nonce is not found in the smart table.
+    public fun get_incoming_nonce_from_bridge_transfer_id(bridge_transfer_id: vector<u8>): u64 acquires SmartTableWrapper {
+        let table = borrow_global<SmartTableWrapper<vector<u8>, u64>>(@aptos_framework);
+
+         // Check if the nonce exists in the table
+        assert!(smart_table::contains(&table.inner, bridge_transfer_id), ENONCE_NOT_FOUND);
+
+        // If it exists, return the associated nonce
+        *smart_table::borrow(&table.inner, bridge_transfer_id)
+    }
+}
+
+module aptos_framework::native_bridge_configuration {
+    use std::signer;
+    use aptos_framework::event;
+    use aptos_framework::system_addresses;
+
+    friend aptos_framework::native_bridge;
+
+    /// Error code for invalid bridge relayer
+    const EINVALID_BRIDGE_RELAYER: u64 = 0x1;
+
+    /// Counterparty time lock duration is 24 hours in seconds
+    const COUNTERPARTY_TIME_LOCK_DUARTION: u64 = 24 * 60 * 60;
+    /// Initiator time lock duration is 48 hours in seconds
+    const INITIATOR_TIME_LOCK_DUARTION: u64 = 48 * 60 * 60;
+
+    struct BridgeConfig has key {
+        bridge_relayer: address,
+        initiator_time_lock: u64,
+        counterparty_time_lock: u64,
+    }
+
+    #[event]
+    /// Event emitted when the bridge relayer is updated.
+    struct BridgeConfigRelayerUpdated has store, drop {
```

Comment:
Why do we need to storer the current and new relayers?
And why not all the past relayers?

Reply:
If each event stores the old one and new one relative to that relayer update, then all such events together will show all past relayers.